llvm/lib/Target/AIE/AIEAlternateDescriptors.h

```diff
@@ -31,6 +31,10 @@ class AIEAlternateDescriptors {
   AIEAlternateDescriptors() = default;
   ~AIEAlternateDescriptors() = default;
 
+  const MIAltDescsMap &getAlternateDescriptors() const {
+    return AlternateDescs;
+  }
```

Comment:
Nit: I'd rather avoid exposing too much about the internal data structure unless really needed. After all, `AIEAlternateDescriptors` is meant as a container, so I think we should just define:
```
MIAltDescsMap::const_iterator begin() const { return AlternateDescs.begin(); }
MIAltDescsMap::const_iterator end() const { return AlternateDescs.end(); }
```